### DISCUSSION TITLE
WooCommerce Products Visibility with loop grid

### DISCUSSION THREAD

**Discussion by Cassiopea001** [elementor/elementor#27531] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

Hello,

![Capture d’écran, le 2024-05-28 à 17 54 03](https://github.com/elementor/elementor/assets/171093629/4e50ae1c-6126-4471-a4c0-f637d36c76c7)
I hope that you are doing well.

We use the WooCommerce Products Visibility by User Roles plugin. 
The plugin uses the 'woocommerce_product_is_visible' hook to manage the visibility of products on the shop page.

More precisely : the 'add_action('woocommerce_product_query', array( $this, 'afpvu_custom_pre_get_posts_query' ), 200, 2);
add_filter('woocommerce_product_is_visible', array( $this, 'afpvu_check_visibility_rules' ), 10, 2);' 

Which hook are you using instead or which hook do we need to use to hide the products? 

Thank you in advance for your help.
Please let us know if you need anything,

### Describe the solution you'd like.

which hook do we need to use to hide the products instead of woocommerce_product_is_visible ?

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by Studio21Creative** [comment]
Hi @elementorbot  - Would you be able to help with this question please? We're facing the same issue.
